src/Compilers/CSharp/Portable/Syntax/SyntaxFacts.cs

```diff
@@ -230,6 +239,9 @@ public static bool IsInNamespaceOrTypeContext(ExpressionSyntax node)
                             // context.
                             return ((QualifiedNameSyntax)parent).Left == node;
 
+                        case QualifiedCref:
+                            return ((QualifiedCrefSyntax)parent).Container == node;
```

Comment:
Note: there was one additional type of interest: NameMemberCrefSyntax.  It has a field `TypeSyntax name`, which indicates that it coudl be a 'types-only' context.  However, both the name of hte class, and the docs seem to strongly indicate this is not so.  specifically:

```
    /// <summary>
    /// A MemberCrefSyntax specified by a name (an identifier, predefined type keyword, or an alias-qualified name,
    /// with an optional type parameter list) and an optional parameter list.
    /// For example, "M", "M&lt;T&gt;" or "M(int)".
    /// Also, "A::B()" or "string()".
    /// </summary>
```

Given this, i have not marked it as a types-only contxt.  It feels like this would have been better represented as two separate classes:  `ParameterizedNameCrefSyntax` and `ParameterizedTypeCrefSyntax`, where the latter would only exist so you could write something like `string()`.

Reply:
Note: compiler assumes this must be a namspace or type, as evidenced here:

```c#
        private ImmutableArray<Symbol> BindQualifiedCref(QualifiedCrefSyntax syntax, out Symbol ambiguityWinner, DiagnosticBag diagnostics)
        {
            NamespaceOrTypeSymbol container = BindNamespaceOrTypeSymbolInCref(syntax.Container);
```